src/Data/List/Properties.agda

```diff
@@ -496,6 +496,22 @@ concat-map {f = f} xss = begin
   foldr (λ ys → map f ys ++_) [] xss         ≡⟨ sym (foldr-fusion (map f) [] (map-++-commute f) xss) ⟩
   map f (concat xss)                         ∎
 
+concat-++ : (xss yss : List (List A)) → concat xss ++ concat yss ≡ concat (xss ++ yss)
+concat-++ [] yss = refl
+concat-++ ([] ∷ xss) yss = concat-++ xss yss
+concat-++ ((x ∷ xs) ∷ xss) yss = cong (x ∷_) (concat-++ (xs ∷ xss) yss)
+
+concat-concat : concat {a} {A} ∘ map concat ≗ concat ∘ concat
```

Comment:
I'm surprised `{a}` is necessary here, shouldn't it be inferrable from `{A}`?

Reply:
Indeed it is, I've made that `{_}` now in `concat-concat` and `concat-[-]`

Reply:
The usually idiomatic way of doing this instead of having `{_} {A}` would be to instead write `{A = A}`

Reply:
Amended